drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -282,6 +343,33 @@ static int siwx91x_status(const struct device *dev, struct wifi_iface_status *st
 	return 0;
 }
 
+static int siwx91x_mode(const struct device *dev, struct wifi_mode_info *mode)
+{
+	struct siwx91x_dev *sidev = dev->data;
+	sl_wifi_interface_t interface;
+	int ret;
+
+	__ASSERT(mode, "mode cannot be NULL");
+
+	if (mode->oper == WIFI_MGMT_SET) {
+		ret = siwx91x_nwp_reboot(mode->mode);
+		if (ret) {
+			return -ETIMEDOUT;
```

Comment:
`ret` can also be `-EINVAL`. Does it make sense to return `-ETIMEDOUT` in this case? Just `return ret`.